frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/components/OktaConditionalAccessModal/OktaConditionalAccessModal.tsx

```diff
@@ -0,0 +1,396 @@
+          "Assertion Consumer Service URL must be a valid URL with http or https scheme and a host";
+      }
+    } catch {
+      errs[OKTA_ACS_URL] =
+        "Assertion Consumer Service URL must be a valid URL with http or https scheme and a host";
+    }
+  }
+
+  // Audience URI validation
+  if (!formData[OKTA_AUDIENCE_URI] || !formData[OKTA_AUDIENCE_URI].trim()) {
+    errs[OKTA_AUDIENCE_URI] = "Audience URI must be present";
+  } else if (formData[OKTA_AUDIENCE_URI].length > maxURLLength) {
+    errs[
+      OKTA_AUDIENCE_URI
+    ] = `Audience URI must be ${maxURLLength} characters or less`;
+  }
+
+  // Certificate validation
+  if (!formData[OKTA_CERTIFICATE] || !formData[OKTA_CERTIFICATE].trim()) {
+    errs[OKTA_CERTIFICATE] = "Certificate must be present";
+  } else if (formData[OKTA_CERTIFICATE].length > maxCertLength) {
+    errs[
+      OKTA_CERTIFICATE
+    ] = `Certificate must be ${maxCertLength} characters or less`;
+  }
+
+  return errs;
+};
+
+export interface IOktaConditionalAccessModalProps {
+  onCancel: () => void;
+  onSuccess: (updatedConfig: IConfig) => void;
+}
+
+const OktaConditionalAccessModal = ({
+  onCancel,
+  onSuccess,
+}: IOktaConditionalAccessModalProps) => {
+  const { renderFlash } = useContext(NotificationContext);
+  const { config } = useContext(AppContext);
+
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [formData, setFormData] = useState<IFormData>({
+    [OKTA_IDP_ID]: "",
+    [OKTA_ACS_URL]: "",
+    [OKTA_AUDIENCE_URI]: "",
+    [OKTA_CERTIFICATE]: "",
+  });
+  const [formErrors, setFormErrors] = useState<IFormErrors>({});
+  const [certFile, setCertFile] = useState<File | null>(null);
+  const [appleProfile, setAppleProfile] = useState<string>("");
+
+  // Fetch Apple profile on mount
+  useEffect(() => {
+    const fetchAppleProfile = async () => {
+      try {
+        const profileText = await conditionalAccessAPI.getIdpAppleProfile();
+        setAppleProfile(profileText);
+      } catch (e: any) {
+        // When responseType is "text", error responses come back as JSON strings
+        // that need to be parsed manually
+        let errorReason = "";
+        try {
+          if (e.data && typeof e.data === "string") {
+            const parsedError = JSON.parse(e.data);
+            errorReason = parsedError.errors?.[0]?.reason || "";
+          } else {
+            errorReason = getErrorReason(e);
+          }
+        } catch {
+          errorReason = getErrorReason(e);
+        }
+
+        const message = errorReason
+          ? `Failed to load Apple profile: ${errorReason}`
+          : "Failed to load Apple profile.";
+        renderFlash("error", message);
+      }
+    };
+    fetchAppleProfile();
+  }, [renderFlash]);
+
+  const onSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+
+    const errs = validate(formData);
+    if (Object.keys(errs).length > 0) {
+      setFormErrors(errs);
+      return;
+    }
```

Comment:
Maybe prevent the API call altogether if somehow `config` is still undefined to prevent any
unintentional unsetting of the entra value:

```
if (!config) {
  return
}
```